src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/ContextQuery/VisualBasicSyntaxContext.vb

```diff
@@ -147,6 +151,32 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
             Return False
         End Function
 
+        Private Shared Function ComputeIsInheritanceRequiringClassContext(targetToken As SyntaxToken) As Boolean
+            If targetToken.IsChildToken(Of InheritsStatementSyntax)(Function(n) n.InheritsKeyword) AndAlso targetToken.HasAncestor(Of ClassBlockSyntax)() Then
+                Return True
+            End If
+
+            Return targetToken.IsChildToken(Of QualifiedNameSyntax)(Function(n) n.DotToken) AndAlso
+                   targetToken.Parent?.FirstAncestorOrSelf(Of InheritsStatementSyntax) IsNot Nothing AndAlso
+                   targetToken.HasAncestor(Of ClassBlockSyntax)()
```

Comment:
This doesn't seem right. For example of you're in the type argument of a generic type that's in an inherits clause

Reply:
Our restrictions are not valid inside generic parameters. `class C : IEnumerable<Action>` is a valid code ([proof](https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQAwAIpwHQGED2ANoQKYDGALgJb4B2AzgNzJqZwAszSLAzJjOlzoQbHgB4sMAHzIA3snSLMfAJIBRWgFcAtiQBOAQwr49EuNPQBxEhQ079RkwAoAlAqXykS7+goALPXwAd3RaEhCAOXwKFW0AB1JdWgoSYDUADzISOOo6Vy5vAF9kd0V1LV1DYz1RbGtbCodq11L0Tx8lf0CQsMjo2ISSJJS0zOzc2nzW4qRCoA)) although `class C : Action` isn't. So it is just easier to not consider such context as valid for our case rather than filter such cases later when filtering symbols

Reply:
i don't understand what you mean.  i don't see anything in your check that distinguishes these cases.

Reply:
>  i don't see anything in your check that distinguishes these cases.

SyntaxTreeExtensions, line 3034:
`(possibleBaseTypeSyntax is null || possibleBaseTypeSyntax.Type is QualifiedNameSyntax { Right: IdentifierNameSyntax });`

This check filters only cases like `class C: $$` (first part) and `class C : System.$$` (second one). Both `class C : IEnumerable<$$` and `class C : System.Collections.Generic.IEnumerable<$$` are filtered here. First example because `IEnumerable<` is not a quilified name, scond - because in quilified name `System.Collections.Generic.IEnumerable<` the right side (`IEnumerable<`) is not an identifier name

Reply:
what if you are here: `class C : System.Collections.Generic.IEnum$$erable<`?

Reply:
also, the above code is VB, but you seem to be showing a bunch of cases that are all C#

Reply:
Implemented handling of `class C : System.Collections.Generic.IEnum$$erable<`. It was already done in VB